homeassistant/components/smappee_official/__init__.py

```diff
@@ -0,0 +1,96 @@
+"""Support for the Smappee API."""
+import logging
+
+from pysmappee import Smappee
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_PASSWORD,
+    CONF_PLATFORM,
+    CONF_USERNAME,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+from .const import DATA_CLIENT, DOMAIN, MIN_TIME_BETWEEN_UPDATES, SMAPPEE_COMPONENTS
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Inclusive(CONF_CLIENT_ID, "Server credentials"): cv.string,
+                vol.Inclusive(CONF_CLIENT_SECRET, "Server credentials"): cv.string,
```

Comment:
Normally, we no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

We instead just want to use a config flow to create config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

If this integration would use our oauth2 helper to accomplish an oauth2 flow there's an exception to this rule, where we instead require the client id and secret to only be set in the config yaml and not in the config flow.

What kind of authentication mechanism does this integration use? Could it benefit from using our oauth2 helper?
https://developers.home-assistant.io/docs/config_entries_config_flow_handler#configuration-via-oauth2

Reply:
The CONFIG_SCHEMAhas been removed in commit [#f713645](https://github.com/home-assistant/core/pull/36445/commits/f713645c1395565c62dc603869ab1e95e557e859). Not sure however if this is enough to restrict yaml config.

Reply:
Could we use our oauth2 helper or is that not possible?

Reply:
Can we schedule the oauth2 helper for another PR?

Reply:
It would be a breaking change for users since the yaml config schema would change. We should avoid breaking changes.

Reply:
Can you point out the changes needed without using the oauth2 helper?

Reply:
You still haven't answered my question if we can use the oauth2 helper. If we can do that, we should use it.

Reply:
Yes, we can look into using the oauth2 helper. I'm following the instructions on https://developers.home-assistant.io/docs/config_entries_config_flow_handler#configuration-via-oauth2 and will reach out to hello@... . What other steps do I need to follow to refactor the integration?

Reply:
Our integration combines data obtained through both our API and MQTT broker, so the auth helper will only cover the API part. Is our integration a good one to use your oauth2 helper?

Reply:
I think so. We can add more config flow steps in the config flow and/or ask the user for further details as needed.

What are the requirements to connect to the MQTT broker?

Reply:
The example code generated by the scaffold script should take you most of the way to a working integration. Look at the existing integrations that use the oauth2 helper for the details, eg the somfy integration.

https://github.com/home-assistant/core/tree/dev/homeassistant/components/somfy

Reply:
@bsmappee no need to email hello@ (I did see your email), we can point you to places here.

Reply:
> I think so. We can add more config flow steps in the config flow and/or ask the user for further details as needed.
> 
> What are the requirements to connect to the MQTT broker?

The client_id and client_secret in the configuration.yaml file together with username and password in the account linking service is enough to setup the integration. So no additional config flow steps needed.